cmd/terramate/e2etests/core/script_info_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2023 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package core_test
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	. "github.com/terramate-io/terramate/cmd/terramate/e2etests/internal/runner"
+
+	"github.com/terramate-io/terramate/test/sandbox"
+)
+
+func TestScriptInfo(t *testing.T) {
+	t.Parallel()
+
+	layout := []string{"d:stacks"}
+
+	addStackWithScripts := func(path string, scripts []string) {
+		layout = append(layout,
+			fmt.Sprintf("s:%s", path))
+
+		var content string
+
+		for _, name := range scripts {
+			content += fmt.Sprintf(`script "%s" {
+				description = "%s at /%s"
+				job {
+					commands = [
+						["echo", "1a"],
+						["echo", "1b"]
+					]
+				}
+				job {
+					command = ["echo", "2"]
+				}
+			}
+			`, name, name, path)
+		}
+
+		layout = append(layout,
+			fmt.Sprintf(`f:%s/script.tm:%s`, path, content))
+	}
+
+	mkExpected := func(name, path, loc string, stacks []string) string {
+		stackstr := ""
+		for _, s := range stacks {
+			stackstr += fmt.Sprintf("\n  %s", s)
+		}
+
+		return fmt.Sprintf(`Definition: /%s/script.tm:%s
+Description: %s at /%s
+Stacks:%s
+Jobs:
+  * ["echo","1a"]
+    ["echo","1b"]
+  * ["echo","2"]
+
+`, path, loc, name, path, stackstr)
+	}
+
+	addStackWithScripts("stacks/a", []string{"deploy", "other"})
+	addStackWithScripts("stacks/a/a1", []string{"other2"})
+	addStackWithScripts("stacks/a/a1/a2", []string{"deploy"})
+	addStackWithScripts("stacks/b", []string{"deploy", "other"})
+	addStackWithScripts("stacks/bb", []string{"other3"})
+
+	s := sandbox.New(t)
+	s.BuildTree(layout)
```

Comment:
it's minor but I feel like generating the buildTree instructions makes the test hard to read. I think it's a good abstraction for table test, where the programmer have a small, terse language for building dirs and files, etc.
But when you can do it programmatically (not a table test), then I feel like using:
```go
s := sandbox.New(t)
test.WriteFile(t, s.RootDir(), "stacks/a/script.tm", scriptAContent)
test.WriteFile(t, s.RootDir(), "stacks/b/script.tm", scriptBContent)
```
makes it easier to understand the test.

We can improve this later though. Wdyt?